src/model/TaskManager.cpp

```diff
@@ -110,101 +120,140 @@ std::vector<Core::TaskEntity> TaskManager::getTaskWithSubtasks(const Core::TaskI
+        if (!ChildrenOf(id).empty() && !deleteChildren) {
+            result.set_status(Core::ModelInquiryResult_Status_HAS_CHILDREN);
+        } else {
+            std::optional<Core::TaskID> ancestor = ParentOf(id);
+            if (ancestor && ToBool(IsPresent(*ancestor)))
+                RemoveChild(*ancestor, id);
+
+            for (auto const &ch: ChildrenOf(id))
+                Delete(ch, true);
+
+            tasks_.erase(id);
+
+            result.set_allocated_id(std::make_unique<Core::TaskID>(id).release());
+        }
     } else {
-        return {ActionResult::Status::ID_NOT_FOUND, id};
+        result.set_status(Core::ModelInquiryResult_Status_ID_NOT_FOUND);
     }
+
+    return result;
 }
 
-ActionResult TaskManager::Edit(const Core::TaskID &id, const Core::Task &t) {
+Core::ModelInquiryResult TaskManager::Edit(const Core::TaskID &id, const Core::Task &t) {
+    Core::ModelInquiryResult result;
     auto it = tasks_.find(id);
+
     if (it != tasks_.end()) {
         it->second.first = t;
-        return {ActionResult::Status::SUCCESS, id};
+        result.set_allocated_id(std::make_unique<Core::TaskID>(id).release());
     } else {
-        return {ActionResult::Status::ID_NOT_FOUND, id};
+        result.set_status(Core::ModelInquiryResult_Status_ID_NOT_FOUND);
     }
+
+    return result;
 }
 
-ActionResult TaskManager::Complete(const Core::TaskID &id) {
+Core::ModelInquiryResult TaskManager::Complete(const Core::TaskID &id) {
+    Core::ModelInquiryResult result;
     auto it = tasks_.find(id);
+
     if (it != tasks_.end()) {
         it->second.first.set_is_complete(true);
+        for (auto const &ch : ChildrenOf(id))
+            Complete(ch);
+
+        result.set_allocated_id(std::make_unique<Core::TaskID>(id).release());
     } else {
-        return {ActionResult::Status::ID_NOT_FOUND, id};
+        result.set_status(Core::ModelInquiryResult_Status_ID_NOT_FOUND);
     }
 
-    for (auto const &ch : ChildrenOf(id))
-        Complete(ch);
-    return {ActionResult::Status::SUCCESS, id};
+    return result;
 }
 
-ActionResult TaskManager::Uncomplete(const Core::TaskID &id) {
+Core::ModelInquiryResult TaskManager::Uncomplete(const Core::TaskID &id) {
+    Core::ModelInquiryResult result;
     auto it = tasks_.find(id);
+
     if (it != tasks_.end()) {
         it->second.first.set_is_complete(false);
+        for (auto const &ch : ChildrenOf(id))
+            Uncomplete(ch);
+
+        result.set_allocated_id(std::make_unique<Core::TaskID>(id).release());
     } else {
-        return {ActionResult::Status::ID_NOT_FOUND, id};
+        result.set_status(Core::ModelInquiryResult_Status_ID_NOT_FOUND);
     }
-    for (auto const &ch : ChildrenOf(id))
-        Uncomplete(ch);
-    return {ActionResult::Status::SUCCESS, id};
+
+    return result;
 }
 
-ActionResult TaskManager::IsPresent(const Core::TaskID &id) const{
+Core::ModelInquiryResult TaskManager::IsPresent(const Core::TaskID &id) const{
```

Comment:
Methods with prefix `Is` should return `bool`.

Reply:
I will change the output parameter in a separate PR